ql/src/java/org/apache/hadoop/hive/ql/parse/RowResolver.java

```diff
@@ -81,7 +81,10 @@ public RowResolver() {
   public void putExpression(ASTNode node, ColumnInfo colInfo) {
     String treeAsString = node.toStringTree();
     expressionMap.put(treeAsString, node);
-    put("", treeAsString, colInfo);
+    if (!putInternal("", treeAsString, colInfo)) {
+      return;
+    }
+    colInfo.setAlias(treeAsString);
```

Comment:
Why not invert the criteria and move colInfo.setAlias(treeAsString); inside the if. I think that would increase code radability.

Reply:
Inverted since only one line moved into the If block but I think it is a question code style. I try to avoid moving and indenting several lines into if blocks when early return is an option.

Reply:
Ok